node_modules/oae-core/manageaccess/js/manageaccess.js

```diff
@@ -172,12 +307,38 @@ define(['jquery', 'oae.core', 'jquery.autosuggest'], function($, oae) {
          * @param  {Object[]}    autoSuggestMembers    Trimmed member object as used by the members feed containing all properties necessary to render a list item in the infinite scroll
          */
         var addNewMembers = function(autoSuggestMembers) {
-            $.each(autoSuggestMembers, function(i, newMember) {
+            var membersPartition = _.partition(autoSuggestMembers, function(member) {
+                return (member.profile.resourceType === 'email');
+            });
+
+            var newInvitations = _.first(membersPartition);
+            var newMembers = _.last(membersPartition);
+
+            // Keep track of all members updates, as well as all members we've searched for by
+            // email in order to create validated share targets
+            _.each(newMembers, function(newMember) {
+                if (newMember.email) {
+                    userIdValidations[newMember.profile.id] = newMember.email;
+                }
+
                 membersUpdates[newMember.profile.id] = newMember.role;
                 pendingMembers[newMember.profile.id] = true;
             });
 
-            infinityScroll.prependItems({'results': autoSuggestMembers});
+            // Keep track of all invitations updates
+            _.each(newInvitations, function(newInvitation) {
+                invitationsUpdates[newInvitation.profile.id] = newInvitation.role;
+                pendingInvitations[newInvitation.profile.id] = true;
+            });
+
+            // Update the members infinite scroll and the invitations list
+            if (!_.isEmpty(newInvitations)) {
+                updateInvitations();
```

Comment:
In essence I think we should, though only if you're OK with how ugly it would be. I think we'd have to have one template for the invitation list container, one template for each individual item, then manually iterate / append items to build the list, and do prepends for individual additions afterward.

Do you think it's worth it?